File: 15-Algorithms_dynamic_programming/memoization2.js
const addTo80 = n => n + 80;

const memoizedAddTo80 = () => {
  let cache = {};

  return n => {
    if (n in cache) return cache[n];

    console.log('long time');
    return (cache[n] = n + 80);
  };
};

const memoized = memoizedAddTo80();
console.log(1, memoized(6));
console.log(2, memoized(6));

File: 15-Algorithms_dynamic_programming/memoization2.py
def add_to_80(n):
    return n + 80


def memoized():
    cache = {}

    def memo(n):
        if n in cache:
            return cache[n]

        print("long time")
        cache[n] = n + 80
        return cache[n]

    return memo


mem = memoized()
print(mem(6))
print(mem(6))
